sw/device/silicon_creator/mask_rom/romextimage_ptrs.h

```diff
@@ -0,0 +1,52 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_ROM_EXT_MANIFEST_PTR_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_ROM_EXT_MANIFEST_PTR_H_
+
+#include "sw/device/silicon_creator/lib/manifest.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+static_assert((TOP_EARLGREY_EFLASH_SIZE_BYTES % 2) == 0,
+              "Flash size is not divisible by 2");
+
+#ifndef OT_OFF_TARGET_TEST
+/**
+ * Returns a pointer to the manifest of the ROM_EXT image stored in flash
+ * slot A.
+ *
+ * @return Pointer to the manifest of the ROM_EXT image in slot A.
+ */
+inline const manifest_t *romextimage_slot_a_manifest_ptr_get(void) {
+  return (const manifest_t *)TOP_EARLGREY_EFLASH_BASE_ADDR;
+}
+
+/**
+ * Returns a pointer to the manifest of the ROM_EXT image stored in flash
+ * slot B.
+ *
+ * @return Pointer to the manifest of the ROM_EXT image in slot B.
+ */
+inline const manifest_t *romextimage_slot_b_manifest_ptr_get(void) {
+  return (const manifest_t *)(TOP_EARLGREY_EFLASH_BASE_ADDR +
```

Comment:
I actually wonder if it's appropriate for this to be volatile... I don't want us using volatile pointers though so maybe we should actually be using MMIO here...?

Reply:
I don't think we need volatile here because the value pointed to never changes during the execution of mask rom. I'd love to hear why you wondered if this should be volatile though.

Reply:
I was worried about LLVM going and inferring something out of the linker script during LTO, but I chatted w some compiler folks who say this is not an optimization LLVM will do. So this is fine. =)